test/functional/feature_assumeutxo.py

```diff
@@ -337,6 +338,22 @@ def assert_only_network_limited_service(self, node):
         assert 'NETWORK' not in node_services
         assert 'NETWORK_LIMITED' in node_services
 
+    @contextlib.contextmanager
+    def assert_disk_cleanup(self, node, assumeutxo_used):
+        """
+        Ensure an assumeutxo node is cleaning up the background chainstate
+        """
+        msg = []
+        if assumeutxo_used:
+            # Check that the snapshot actually existed before restart
+            assert (node.datadir_path / node.chain / "chainstate_snapshot").exists()
```

Comment:
nit: `node.datadir_path / node.chain / "chainstate_snapshot"` could be extracted to obviate that it's checked multiple times

Reply:
Will do if I have to retouch